libs/langchain/langchain/storage/file_system.py

```diff
@@ -36,14 +36,25 @@ class LocalFileStore(ByteStore):
 
     """
 
-    def __init__(self, root_path: Union[str, Path]) -> None:
+    def __init__(
+            self,
+            root_path: Union[str, Path],
+            chmod_file: Optional[int] = None,
```

Comment:
can we keep the same mode for the file and dir?

Reply:
@eyurtsev - this is what I originally wanted to do, but the conventions for file and directory permissions are too different. For example, directories have access bits, sticky bits, and other characteristics that do not apply to files. In addition, there are subtle differences between file and directory bits across platforms.

See the `stat.S_*` flag documentation [located here](https://docs.python.org/3/library/stat.html#stat.S_ISUID) for more details.

I thought about trying to do a "best guess" derivation at directory permission bits based on the specified file permission bits, but (1) someone will want something different than this guess, and (2) such hidden-magic derivation goes against the Pythonic way of keeping things explicit and obvious.